wordpress-mobile/WordPress-iOS
# Extract notifications feature branch merge

This is a quite big PR that merges the `Extract Notification Interactions` feature branch into develop.

Most of the new classes and files are part of a refactor of `NotificationBlock` as `FormattableContent`, where we tried to make it independent of `Notification` specific code, and reusable.

Another section of the new code is implementing this `FormattableContent` on activity log.
And many of the new files are part of the test suit.

All these changes where already reviewed and tested on many different previous pull requests, so a smoke test should be enough to be approved.

We decided to keep the feature flag around for a release cycle in case we need to revert changes, all the old code is still present under the feature flag.

There is [a PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/9878) to remove the feature flag and all these extra code ready to be merged when necessary.

To test:
- Check that Notifications work as it was before.
- Check that Activity Log events show formatted content and the routes work.
